src/OpenLoco/src/Ui/Windows/Construction/Construction.h

```diff
@@ -172,18 +173,23 @@ namespace OpenLoco::Ui::Windows::Construction
         void setTrackOptions(const uint8_t trackType);
         void setDisabledWidgets(Window* self);
         void createConstructionWindow();
-        void refreshAirportList(uint8_t* stationList);
-        void refreshDockList(uint8_t* stationList);
-        void refreshStationList(uint8_t* stationList, uint8_t trackType, TransportMode transportMode);
-        void refreshBridgeList(uint8_t* bridgeList, uint8_t trackType, TransportMode transportMode);
-        void refreshModList(uint8_t* modList, uint8_t trackType, TransportMode transportMode);
         void sub_4A3A50();
-        void refreshSignalList(uint8_t* signalList, uint8_t trackType);
         void setNextAndPreviousTrackTile(const TrackElement& elTrack, const World::Pos2& pos);
         void setNextAndPreviousRoadTile(const RoadElement& elRoad, const World::Pos2& pos);
         bool isPointCloserToNextOrPreviousTile(const Point& point, const Viewport& viewport);
         void previousTab(Window* self);
         void nextTab(Window* self);
+
+        template<uint32_t NewCapacity, uint32_t LegacyCapacity>
+        void copyToLegacyList(const sfl::static_vector<uint8_t, NewCapacity>& sflType, uint8_t (&legacyList)[LegacyCapacity])
+        {
+            static_assert(LegacyCapacity >= NewCapacity);
```

Comment:
Don't we require LegacyCapacity to be always MaxObjects + 1 ? So the assertion should probably be `LegacyCapacity > NewCapacity` since we always write to `legacyList[sflType.size()]` if they were equal then it would be out of bounds.